### PR TITLE
Use is_trusted bool in insert_shreds() instead manually adjusting root

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
    The test_duplicate_with_pruned_ancestor test needs to get around a
    limitation where the shreds with a parent older than the latest root are
    discarded. The previous approach manually adjusted the root value in the
    blockstore; this is not ideal in that it is fiddling with the inner
    working of Blockstore.
    
#### Summary of Changes
    So, use the is_trusted argument in Blockstore::insert_shreds(); setting
    is_trusted=true bypasses the sanity checks (including the parent >=
    latest root check).

### CODE REVIEW

File: ledger/src/blockstore.rs [hunk 1]
```diff
@@ -3256,11 +3256,6 @@ impl Blockstore {
         Ok(())
     }
 
-    /// For tests
-    pub fn set_last_root(&mut self, root: Slot) {
```
Comment (steviez) [inline ledger/src/blockstore.rs:3260]: @CriesofCarrots - How do you feel about me ripping this (technically `pub`) `Blockstore` function out? It was added for a local cluster test, and this PR obsoletes it

File: local-cluster/tests/local_cluster.rs [hunk 2]
```diff
@@ -4714,11 +4715,16 @@ fn test_duplicate_with_pruned_ancestor() {
             remove_tower(&our_node_ledger_path, &majority_pubkey);
         }
 
-        // Copy minority fork. Rewind our root so that we can copy over the purged bank
+        // Copy minority fork to our blockstore
+        // Set trusted=true in blockstore copy to override parent >= latest root check
```
Comment (AshwinSekar) [inline local-cluster/tests/local_cluster.rs:4719]: nit:
```suggestion
        // Set trusted=true in blockstore copy to override parent >= latest root check, which would have prevented the copy of the pruned fork
```

File: ledger/src/blockstore.rs (hunk 1 above)
Comment (CriesofCarrots) [reply ledger/src/blockstore.rs:3260]: I don't mind ripping a test-only method out.

**Review by AshwinSekar — DISMISSED** [review]
thanks for this, a much better solution than my hack! just a minor nit on a comment

**Review by AshwinSekar — APPROVED** [review]

**Comment by steviez** [comment]
I was initially a little concerned in that one of the local-cluster CI steps took 3 retries to succeed. However, looking at the logs for those 3 failures, all 3 failed at the same spot on a different test:
```
thread 'test_optimistic_confirmation_violation_without_tower' panicked at local-cluster/tests/local_cluster.rs:3382:13:

Violation expected because of removed persisted tower!

test test_optimistic_confirmation_violation_without_tower has been running for over 60 seconds

test test_optimistic_confirmation_violation_without_tower ... FAILED
```
This is not the test whose behavior I altered; `test_optimistic_confirmation_violation_without_tower` does call `copy_block()` as well, but it has a value of `false` set which matches the pre-existing behavior. Namely, [this false](https://github.com/solana-labs/solana/pull/34010/files#diff-5e9f940ea065adfd3066ef0d8ef0cfb5029b5ba478d96fb484b36954e464505cR3288) matches the [previously hard-coded false in copy_blocks()](https://github.com/solana-labs/solana/pull/34010/files#diff-82c7402cbce63ec9670dd82e5b9e83e74e9b7f67cbfce9176e52cdba13a077acL180).

So, it would appear that this test was previously flaky, and my PR does not alter the flaky test so I think I can ok to submit. Do you agree with my conclusion @AshwinSekar; also CC @yihau as I know you've been keeping an eye on flaky tests

**Comment by AshwinSekar** [comment]
yeah that's a known flaky test. i was unable to reproduce it locally and then it quickly got buried in my todo. I'll take a look again at it next week https://github.com/solana-labs/solana/issues/29221

**Pull request merged**